models.py

```diff
@@ -21,6 +21,8 @@
 from sqlalchemy.event import listens_for
 from sqlalchemy.orm import relationship
 from sqlalchemy.sql import func as sqlfunc
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy import func
```

Comment:
This was my bad for telling you to add this import, as I didn't realize we already imported `func` on line 23 (we just rename it to `sqlfunc`). Could you remove this import and replace your use of `func` on line 174 with `sqlfunc`?

Once that's done, remove the `[WIP]` from this PR's title, and it'll be ready to review.